syntax = "proto3";
import "protobufs/Common/User.proto";

message AchievementListRequest {
    string user_name = 1;
}

message AchievementListResponse {
    repeated Achievement achievements = 1;
}

message AchievementRequest{
    string title = 1;
    string user_name = 2;
}

message AchievementResponse{
    Achievement item = 1;
}

message UpdateRequest{
    User user_to_update = 1;
    Achievement new = 2;
}

message UpdateResponse{
    bool success = 1;
}

service AchievementsController {
    rpc GetAchivementList (AchievementListRequest) returns (AchievementListResponse);
    rpc GetAchievement (AchievementRequest) returns (AchievementResponse);
    rpc UpdateAchievement (UpdateRequest) returns (UpdateResponse);
}